src/common/redis_module/ray_redis_module.cc

```diff
@@ -1098,25 +1109,40 @@ int TaskTableTestAndUpdate_RedisCommand(RedisModuleCtx *ctx,
     return RedisModule_ReplyWithError(ctx, "Found invalid scheduling state.");
   }
   long long test_state_bitmask;
-  int status = RedisModule_StringToLongLong(argv[2], &test_state_bitmask);
+  int status = RedisModule_StringToLongLong(test_state, &test_state_bitmask);
   if (status != REDISMODULE_OK) {
     RedisModule_CloseKey(key);
     return RedisModule_ReplyWithError(
         ctx, "Invalid test value for scheduling state");
   }
 
-  bool updated = false;
+  bool update = false;
   if (current_state_integer & test_state_bitmask) {
-    /* The test passed, so perform the update. */
-    RedisModule_HashSet(key, REDISMODULE_HASH_CFIELDS, "state", state,
-                        "local_scheduler_id", argv[4], NULL);
-    updated = true;
+    update = true;
+    if (test_local_scheduler) {
+      /* A test local scheduler ID was provided. Test whether it is equal to
+       * the current local scheduler ID before performing the update. */
+      RedisModuleString *test_local_scheduler_id = argv[5];
+      if (RedisModule_StringCompare(current_local_scheduler_id,
```

Comment:
Can we do this comparison before the `if (current_state_integer & test_state_bitmask) {` line? It seems a little confusing to start with `update = false`, then set `update = true`, and then set it back to `update = false`.

Reply:
Ah yes, let me clean up this section...